src/ADAL.PCL/Internal/Telemetry/DefaultEvent.cs

```diff
@@ -0,0 +1,119 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.IdentityModel.Clients.ActiveDirectory
+{
+    internal class DefaultEvent : EventsBase
+    {
+        internal Dictionary<string, string> EventDictitionary = new Dictionary<string, string>();
+
+        static DefaultEvent()
+        {
+            ApplicationName = PlatformPlugin.PlatformInformation.GetApplicationName();
+
+            ApplicationVersion = PlatformPlugin.PlatformInformation.GetApplicationVersion();
+
+            SdkVersion = AdalIdHelper.GetAdalVersion();
+
+            SdkPlatform = AdalIdHelper.GetAssemblyFileVersion();
+
+            DeviceId =
+                PlatformPlugin.CryptographyHelper.CreateSha256Hash(PlatformPlugin.PlatformInformation.GetDeviceId());
+        }
+
+        internal DefaultEvent()
+        {
+            SetEvent(EventConstants.ApplicationName, ApplicationName);
+
+            SetEvent(EventConstants.ApplicationVersion, ApplicationVersion);
+
+            SetEvent(EventConstants.SdkVersion, SdkVersion);
+
+            SetEvent(EventConstants.SdkPlatform, SdkPlatform);
+
+            SetEvent(EventConstants.DeviceId, DeviceId);
+        }
+
+        internal string ClientId { get; set; }
+
+        internal static string ApplicationName { get; set; }
+
+        internal static string ApplicationVersion { get; set; }
+
+        internal static string SdkPlatform { get; set; }
+
+        internal static string SdkVersion { get; set; }
+
+        internal static string DeviceId { get; set; }
+
+        internal Guid CorrelationId { get; set; }
+
+        internal override void SetEvent(string eventName, string eventParameter)
+        {
+            if (!string.IsNullOrEmpty(eventParameter) && !string.IsNullOrEmpty(eventName))
+            {
+                EventDictitionary[eventName] = eventParameter;
+            }
+        }
+
+        internal void SetEvent(string eventName, bool eventParameter)
+        {
+            EventDictitionary[eventName] = eventParameter.ToString();
```

Comment:
I just left a comment about setEvent in my previous iteration. you have the same bug in the very next function. I am hoping you pay more attention to code around specially overloads and overrides.